### PR TITLE
fix default suspension of Deployment/StatefulSet with Kueue-managed owners

### PR DESCRIPTION
**Pull request by dgrove-oss**
#### What type of PR is this?

/kind feature
#### What this PR does / why we need it:

Adjusts default suspension logic in Deployment/StatefulSet webhooks to correctly handle the case where they
are children of another Kueue-managed Kind.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

Based on a [suggestion in abandoned PR](https://github.com/kubernetes-sigs/kueue/pull/3778#issuecomment-2530716162) to allow us to avoid needing a full-blown replicaset integration.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by dgrove-oss** [comment]
/hold
I did initial testing with some e2e scenarios with appwrappers already.  I want to work through all of them to make sure there aren't any surprises.  Except to finish that today.

File: pkg/controller/jobs/pod/pod_webhook.go [hunk 1]
```diff
@@ -177,14 +173,53 @@ func (w *PodWebhook) Default(ctx context.Context, obj runtime.Object) error {
...
+			// function so that it can use the standaloneJob path in the GenericReconciler for its Pods.
+			if owner.Kind == "ReplicaSet" && owner.APIVersion == "apps/v1" {
+				rs := &appsv1.ReplicaSet{}
+				err := w.client.Get(ctx, client.ObjectKey{Name: owner.Name, Namespace: pod.pod.GetNamespace()}, rs)
+				if err != nil {
+					return fmt.Errorf("failed to get replicaset: %w", err)
+				}
+				owner = metav1.GetControllerOf(rs)
+				if owner != nil {
+					if owner.Kind == "Deployment" && owner.APIVersion == "apps/v1" && jobframework.IsIntegrationEnabled("deployment") {
```
Comment (mimowo) [inline pkg/controller/jobs/pod/pod_webhook.go:200]: Can we move this out to the same level where StatefulSet is considered, for symmetry. Then move the comment about the Special case for Deployment down just above the if. I also believe we can eliminate the special cases alltogether by the lookup as suggested in the other comment.

File: pkg/controller/jobs/pod/pod_webhook.go [hunk 2]
```diff
@@ -177,14 +173,53 @@ func (w *PodWebhook) Default(ctx context.Context, obj runtime.Object) error {
...
+						return nil
+					}
+				}
+			}
+			// Special case for StatefulSet integration which intentionally does not register an IsManagingObjectsOwner
+			// function so that it can use the standaloneJob path in the GenericReconciler for its Pods.
+			if owner.Kind == "StatefulSet" && owner.APIVersion == "apps/v1" && jobframework.IsIntegrationEnabled("statefulset") {
```
Comment (mimowo) [inline pkg/controller/jobs/pod/pod_webhook.go:207]: LGTM, but I would prefer less hard-coding here.
- the "statefulset" also does not need to be hard-coded it is the value of `pod.pod.GetLabels()[SuspendedByParentLabelKey]`. 
- the "StatefulSet" and "APIVersion", we can read them from GVK of the integration. Example [here](https://github.com/kubernetes-sigs/kueue/blob/d85a79b81c906d1d17197f8d560ba11d018223ac/pkg/controller/jobs/statefulset/statefulset_controller.go#L47). For that we need to change the function `IsIntegrationEnabled` to `enabledIntegrationGVK()` which would return the GVK if enabled.

**Review by mimowo — COMMENTED** [review]
LGTM overall, the main improvement could be to eliminate the hard-coding of special cases.

**Comment by mimowo** [comment]
/lgtm
/approve
Since this PR was extensively tested manually per https://github.com/kubernetes-sigs/kueue/pull/3803#issuecomment-2533506262 I'm good to go with it. The main concern is the hard-coded lookup of the integrations. I believe this is something we can clean up in a follow up @dgrove-oss.

**Pull request merged**